test/e2e_node_kind_source_map_test.ts

```diff
@@ -0,0 +1,625 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// tslint:disable:no-unused-expression mocha .to.be.empty getters.
+
+import {expect} from 'chai';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+import * as tsickle from '../src/tsickle';
+
+import {compile, getLineAndColumn} from './test_support';
+
+describe('source maps each node with transformer', () => {
+  it('maps at the start of lines correctly', () => {
+    const sources = new Map([[
+      'input.ts', `let x : number = 2;
+      x + 1;
+      let y = {z: 2};
+      y.z;`
+    ]]);
+
+    const {compiledJS, sourceMap} = compile(sources, {useTransformer: true});
+
+    {
+      const {line, column} = getLineAndColumn(compiledJS, 'var /** @type {number} */ x');
+      expect(sourceMap.originalPositionFor({line, column}).line)
+          .to.equal(1, 'variable declaration line');
+      expect(sourceMap.originalPositionFor({line, column}).source)
+          .to.equal('input.ts', 'input file name');
+    }
+    {
+      const {line, column} = getLineAndColumn(compiledJS, 'x + 1');
+      expect(sourceMap.originalPositionFor({line, column}).line).to.equal(2, 'addition line');
+      expect(sourceMap.originalPositionFor({line, column}).source)
+          .to.equal('input.ts', 'input file name');
+    }
+    {
+      const {line, column} = getLineAndColumn(compiledJS, 'y.z');
+      expect(sourceMap.originalPositionFor({line, column}).line).to.equal(4, 'object access line');
+      expect(sourceMap.originalPositionFor({line, column}).source)
+          .to.equal('input.ts', 'input file name');
+    }
+  });
+
+  // TODO(lucassloan): enable when we have a custom transformer for classes
```

Comment:
We typically do not check commented out code in - experience is that it has bad chances to stay correct even in the short term. Can you move these bits to a separate branch in your fork, and just send them out once we're ready?